lib/application/referral/provider.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:clones_desktop/domain/models/api/referral_api.dart';
+import 'package:clones_desktop/domain/models/referral/referral_info.dart';
+import 'package:clones_desktop/infrastructure/referral.repository.dart';
+import 'package:clones_desktop/utils/api_client.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter/foundation.dart';
+
+// Repository provider
+final referralRepositoryProvider = Provider<ReferralRepository>((ref) {
+  final apiClient = ref.watch(apiClientProvider);
+  return ReferralRepositoryImpl(apiClient);
+});
+
+// Referral state notifier
```

Comment:
You should separate notifier and state in `ui/views/xxx/bloc` folder to associate notifier management and UI View
Example : `lib/ui/views/forge_detail`